### PR TITLE
Sa termination

### PR DESCRIPTION
**Pull request by drorhayun**
# Pull Request Template

## Prerequisites

* If there is no issue related to your pull request - open one and assign yourself to it. If you're proposing a solution to an already opened issue - simply assign yourself to it.

## Motivation

*Describe the problem you're trying to solve or the feature you're proposing in addition to a link to the related issue.*

## Approach

_How does this change address the problem?_

## Pull Request status

* [x] Initial implementation
* [ ] Refactoring
* [x] Unit tests
* [x] Integration tests

Feel free to construct the checklist with whatever items seem most reasonable for your change. You could disassemble the Implementation part to even smaller separate checklist items if you're working on something big for example. But do make the effort to provide a checklist of some sort so that the core team as well as the community can have an idea about the progress of your Pull Request.

## Third-party code

If you use third party code with your contribution such as, components, libraries, or snippets make sure to mention that.

Make sure that your contribution complies with our [Contribution guidelines][1].

## Work in Progress label

For as long as development of your Pull Request is still ongoing you MUST label it with a **wip** label as well as prefix the name of the PR with *[WIP]*.

For example: *[WIP] Service brokers API*

[1]: https://github.com/Peripli/service-manager/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: pkg/types/cascade/interface.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+func GetCascadeObject(ctx context.Context, object types.Object) (Cascade, bool) {
+	switch object.GetType() {
+	case types.TenantType:
+		return &TenantCascade{object.(*types.Tenant)}, true
+	case types.PlatformType:
+		return &PlatformCascade{object.(*types.Platform)}, true
+	case types.ServiceBrokerType:
+		return &ServiceBrokerCascade{object.(*types.ServiceBroker)}, true
+	case types.ServiceInstanceType:
+		containerIdKey := ctx.Value(ContainerKey{})
```
Comment (eyalnathan) [inline pkg/types/cascade/interface.go:53]: containerIDKey (capital D)

Comment (eyalnathan) [reply pkg/types/cascade/interface.go:53]: or containerLabelKey?

File: pkg/web/routes.go [hunk 2]
```diff
@@ -53,4 +53,6 @@ const (
 
 	// ProfileURL is the Configuration API base URL path
 	ProfileURL = "/" + apiVersion + "/profile"
+
+	TenantURL = "/" + apiVersion + "/tenant"
```
Comment (eyalnathan) [inline pkg/web/routes.go:57]: shouldn't this be /tenants? all other resource endpoints are using plural

File: storage/postgres/migrations/20200514181500_operations_cascade.up.sql [hunk 3]
```diff
@@ -0,0 +1,7 @@
+BEGIN;
+
+ALTER TABLE operations ADD COLUMN IF NOT EXISTS cascade_root_id varchar(100);
```
Comment (eyalnathan) [inline storage/postgres/migrations/20200514181500_operations_cascade.up.sql:3]: do we want to create index on the new columns? the "cascade_root_it" is used for queries in many flows

Comment (pavelmaliy) [reply storage/postgres/migrations/20200514181500_operations_cascade.up.sql:3]: no sure, we're not looking for specific root_id only in cleanup
maybe it'll be better to index parent_id

**Pull request merged**